src/script/api/script_text.cpp

```diff
@@ -198,9 +198,16 @@ void ScriptText::_GetEncodedText(std::back_insert_iterator<std::string> &output,
 		if (idx >= args.size()) throw Script_FatalError(fmt::format("{}({}): Not enough parameters", name, param_count + 1));
 		ParamCheck &pc = args[idx++];
 		if (pc.owner != this->string) ScriptLog::Warning(fmt::format("{}({}): Consumes {}({})", name, param_count + 1, GetGameStringName(pc.owner), pc.idx + 1));
-		return pc.param;
+		return &pc;
 	};
 	auto skip_args = [&](size_t nb) { idx += nb; };
+	auto output_arg = [&](ParamCheck &p) {
+		if (p.used) return;
+		if (std::holds_alternative<std::string>(*p.param)) fmt::format_to(output, ":\"{}\"", std::get<std::string>(*p.param));
+		if (std::holds_alternative<SQInteger>(*p.param)) fmt::format_to(output, ":{:X}", std::get<SQInteger>(*p.param));
+		if (std::holds_alternative<ScriptTextRef>(*p.param)) fmt::format_to(output, ":{:X}", p.owner);
+		p.used = true;
+	};
```

Comment:
This could capture only `output` I think. Or could even be just a standard function instead of lambda?

Reply:
I can make it a static function with `output` as parameter.

Reply:
I'm also not really sure about member functions everywhere, but hey :)